extensions/azurePublishNew/src/node/provisionService.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IBotProject } from '@botframework-composer/types';
+
+import { getAppRegistrationProvisionService } from './azureResources/appRegistration';
+import { getWebAppProvisionService } from './azureResources/webApp';
+import { getBotChannelProvisionService } from './azureResources/botChannel';
+import { getAzureFunctionsProvisionService } from './azureResources/azureFunctions';
+import { getCosmosDbProvisionService } from './azureResources/cosmosDb';
+import { getLuisAuthoringProvisionService } from './azureResources/luisAuthoring';
+import { getLuisPredictionProvisionService } from './azureResources/luisPrediction';
+import { getBlogStorageProvisionService } from './azureResources/blobStorage';
+import { getQnAProvisionService } from './azureResources/qna';
+import { getAppServiceProvisionService } from './azureResources/servicePlan';
+
+// bot project => candidate resources => select & configure resources => order & provision
+
+export type ProvisionWorkingSet = Record<string, object>;
+
+export type ResourceProvisionService = {
+  getDependencies: () => string[];
+  getRecommendationForProject: (project: IBotProject) => 'required' | 'optional' | 'invalid';
+  provision: <TConfig>(config: TConfig, workingSet: ProvisionWorkingSet) => void;
+};
+
+export type ResourceDefinition = {
+  key: string;
+  text: string;
+  description: string;
+  tier: string;
+  group: string;
+};
+
+export type GetResourcesResult = ResourceDefinition & {
+  required: boolean;
+};
+
+type ProvisionConfig = {
+  key: string;
+};
+
+type ProvisionCredentials = {
+  token: string;
+  graphToken: string;
+  subscriptionId: string;
```

Comment:
I don't think the subscription is part of the credentials.  More likely it would be tenantId.  I would add subscriptionId to the config of the resources that need it, or if everyone needs it add it to the ProvisionConfig.

Reply:
added to ProvisionConfig